### PR TITLE
Introduce insecure option for usage with self-signed certificates

### PR DESCRIPTION
**Pull request by RincewindsHat**

### CODE REVIEW

**Comment by uvu9Ba** [comment]
@RincewindsHat I got following error:

```
# ./test.py --api https://xxxx --username xxx --password xxxxx --mode capacity-usage --insecure

[UNKNOWN] Unexpected Python error: <class 'NameError'> name 'verify' is not defined
  File "./test.py", line 429, in <module>
    sys.exit(main())
  File "./test.py", line 421, in main
    return client.get_capacity_usage().print_and_return()
  File "./test.py", line 151, in get_capacity_usage
    return CapacityUsage(self.request('capacity/usage'))
  File "./test.py", line 117, in request
    response = requests.request(method, request_url, auth=HTTPBasicAuth(self.username, self.password), verify=verify)
```

Also `VERSION` in line 42 should be increased, right?

\\

`request` ist not the only think that makes problems because of self-sign certs. Also `urllib3` will show a warning:
```
/usr/local/lib/python3.6/site-packages/urllib3/connectionpool.py:1020: InsecureRequestWarning: Unverified HTTPS request is being made to host 'cc-t-nsx-manager.cc.cronon.systems'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
  ```
  
I fixed it with:
  ```
import urllib3
urllib3.disable_warnings()
```
(maybe there is a better solution)

**Comment by RincewindsHat** [comment]
hm, try again pls

**Comment by uvu9Ba** [comment]
Nope

with
```
    if args.verify == False:
        import urllib3
        urllib3.disable_warnings()
```

I get
```
# ./test.py --api https://xxxx --username xxxx --password xxxx --mode capacity-usage --insecure

[UNKNOWN] Unexpected Python error: <class 'AttributeError'> 'Namespace' object has no attribute 'verify'
  File "./test.py", line 432, in <module>
    sys.exit(main())
  File "./test.py", line 409, in main
    if args.verify == False:
```


without I get:
```
# ./test.py --api https://xxxx --username xxxx --password xxxx --mode capacity-usage --insecure

[UNKNOWN] Unexpected Python error: <class 'AttributeError'> 'Client' object has no attribute 'verify'
  File "./test.py", line 432, in <module>
    sys.exit(main())
  File "./test.py", line 424, in main
    return client.get_capacity_usage().print_and_return()
  File "./test.py", line 151, in get_capacity_usage
    return CapacityUsage(self.request('capacity/usage'))
  File "./test.py", line 117, in request
    response = requests.request(method, request_url, auth=HTTPBasicAuth(self.username, self.password), verify=self.verify)
```

**Comment by RincewindsHat** [comment]
should have seen those

File: check_vmware_nsxt.py [hunk 1]
```diff
@@ -397,19 +398,24 @@ def parse_args():
 
     args.add_argument('--version', '-V', help='Print version', action='store_true')
 
+    args.add_argument('--insecure', help='Do not verify TLS certificate. Be careful with this option, please', action='store_true', required=False)
```
Comment (uvu9Ba) [inline check_vmware_nsxt.py:401]:
```suggestion
    args.add_argument('--insecure', help='Do not verify TLS certificate. Be careful with this option, please', action='store_false', required=False)
```

Im not a python programmer but I guess it was to be `store_false`, right? :)

After switching this to false it works with `--insecure` switch

**Review by uvu9Ba — CHANGES_REQUESTED** [review]

File: check_vmware_nsxt.py (hunk 1 above)
Comment (RincewindsHat) [reply check_vmware_nsxt.py:401]: not my day :-)
Yes, of course

**Comment by uvu9Ba** [comment]
Now it works like expected :)

Now comes the lame part: adjusting documentation :3

* Adjust `VERSION` in script
* README.md should reflect that this option exists
* Extend Icinga config example:
```
    arguments += {
        "--api" = "$vmware_nsx_api$"
        "--mode" = "$vmware_nsx_mode$"
        "--password" = "$vmware_nsx_password$"
        "--username" = "$vmware_nsx_username$"
        "--insecure" = {
            set_if = "$vmware_nsx_insecure$"
        }
    }
```

Thanks for your effort.

**Comment by RincewindsHat** [comment]
Thank you very much for the help :-)

**Pull request merged**